practices/appeng/adrs/0005-postgresql-as-relational-store.md

```diff
@@ -0,0 +1,87 @@
+# 0005 - Use PostgreSQL as a relational store
+
+**Status:** In Progress
+
+**Date Accepted:** TBD
+
+**Reviewers:** @reggieriser @mikena-truss
+
+## Context
+
+It is typical for our projects to require the persistent storage of data. A relational database is
+often a good choice for this as it allows us to predefine a schema for and relationships
+between data. A database management system (DBMS) can interpret and enforce these rules
+while providing a means to efficiently and safely interact with the data. This ADR documents the
+reasoning behind our recommendation of a preferred DBMS.
+
+## Decision
+
+Use PostgreSQL when you need a relational store.
+
+## Why is this Applicable to the Practice as a Whole
+
+Since the requirement for a relational data store is common, there are advantages of us standardizing
+on one DBMS across the practice. Benefits include:
+
+- ability to transfer knowledge across projects
+- reduced ramp-up time when switching projects
+- reuse of infrasec practices
+- norming on a common SQL/DDL dialect
+
+## When to Not Implement This Decision
+
+- If the project's client/sponsor has an established and/or inflexible existing DBMS requirement
+- If the nature of the data lends itself to a different type of data store (NoSQL database, graph database, etc.)
+- The project is so small, self-contained, or temporary that a lightweight solution like SQLite is more appropriate
+
+## Alternatives Considered
+
+### PostgreSQL
+
+#### Pros
+
+- `+` Mature and full-featured DBMS with a 25-year history
+- `+` Free and open-source
+- `+` Good documentation and user community
+- `+` Advanced JSON support so flexible enough for some unstructured data
+- `+` Supported by most database tools and ORMs
+
+#### Cons
+
+- `-` No guaranteed support (unless you pay for third-party support)
+- `-` Some clients may not be comfortable with an open-source DBMS
+
+### MySQL
+
+#### Pros
+
+- `+` Mature DBMS with a 25-year history
+- `+` Community edition is free and open-source
+- `+` Good documentation and user community
+- `+` Supported by most database tools and ORMs
+
+#### Cons
+
+- `-` No guaranteed support (unless you pay for third-party support)
+- `-` Some clients may not be comfortable with an open-source DBMS
+- `-` Given that MySQL is owned by Oracle, some worry that it may be harder to contribute to or not always be free
+- `-` More advanced features are found in the paid Enterprise Edition
+- `-` ACID support is not as mature
+
+### Commercial Databases (Oracle, SQL Server, DB2, etc.)
+
+#### Pros
+
+- `+` Mature DBMS with long history
+- `+` Traditional paid software backed by a company and a support team
+- `+` Supported by most database tools and ORMs
+
+#### Cons
+
+- `-` Licensing fees and support can be very expensive
+- `-` Not open source so you are reliant on company for fixes
+- `-` Support by ORMs can be spotty
```

Comment:
I do have some experience with Oracle databases, and one con you could consider adding is how difficult it is to search for the syntax of Oracle SQL vs. MySQL or other more popular dialects. "SQL" is everywhere, so any version that's markedly less popular than the others is liable to be drowned out of the results when you're looking for help.

Reply:
And, I think you also brought this up in the server guild meeting Sandy, but also error codes can a pain to figure out with Oracle.